modules/ximgproc/src/deriche_filter.cpp

```diff
@@ -0,0 +1,593 @@
+                    j--;
+                    c1--;
+                    for (j = cols - 3; j >= 0; j--, c1--)
+                        g2[j] = a3*c1[1] + a4*c1[2] + b1*g2[j + 1] + b2*g2[j + 2];
+                    for (j = 0; j<cols; j++, f1++)
+                        *f1 = (float)(g1[j] + g2[j]);
+                }
+            }
+            break;
+            case CV_16S:
+            {
+                short *c1;
+                f1 = ((float*)dst.ptr(0));
+                for (int i = range.start; i<range.end; i++)
+                {
+                    c1 = ((short*)img.ptr(0));
+                    c1 += i*cols;
+                    int j = 0;
+                    g1[j] = (a1 + a2)* *c1;
+                    j++;
+                    c1++;
+                    g1[j] = a1 * c1[0] + a2*c1[j - 1] + (b1)* g1[j - 1];
+                    j++;
+                    c1++;
+                    for (j = 2; j<cols; j++, c1++)
+                        g1[j] = a1 * c1[0] + a2 * c1[-1] + b1*g1[j - 1] + b2*g1[j - 2];
+                    c1 = ((short*)img.ptr(0));
+                    c1 += i*cols + cols - 1;
+                    j = cols - 1;
+                    g2[j] = (a3 + a4)* *c1;
+                    j--;
+                    c1--;
+                    g2[j] = (a3 + a4) * c1[1] + (b1)* g2[j + 1];
+                    j--;
+                    c1--;
+                    for (j = cols - 3; j >= 0; j--, c1--)
+                        g2[j] = a3*c1[1] + a4*c1[2] + b1*g2[j + 1] + b2*g2[j + 2];
+                    for (j = 0; j<cols; j++, f1++)
+                        *f1 = (float)(g1[j] + g2[j]);
+                }
+            }
+            break;
+            default:
+                return;
+            }
+        };
+        ParallelGradientDericheXRows& operator=(const ParallelGradientDericheXRows &) {
+            return *this;
+        };
+    };
+
+    UMat GradientDericheY(UMat op, double alphaDerive, double alphaMean)
+    {
+        Mat tmp(op.size(), CV_32FC(op.channels()));
+        UMat imDst(op.rows, op.cols, CV_32FC(op.channels()));
+        cv::Mat opSrc = op.getMat(cv::ACCESS_RW);
+        cv::Mat dst = imDst.getMat(cv::ACCESS_RW);
+        std::vector<Mat> planSrc;
+        split(opSrc, planSrc);
+        std::vector<Mat> planTmp;
+        split(tmp, planTmp);
+        std::vector<Mat> planDst;
+        split(dst, planDst);
+        for (int i = 0; i < static_cast<int>(planSrc.size()); i++)
+        {
+            if (planSrc[i].isContinuous() && planTmp[i].isContinuous() && planDst[i].isContinuous())
+            {
+                ParallelGradientDericheYCols x(planSrc[i], planTmp[i], alphaDerive);
+                parallel_for_(Range(0, opSrc.cols), x, getNumThreads());
+                ParallelGradientDericheYRows xr(planTmp[i], planDst[i], alphaMean);
+                parallel_for_(Range(0, opSrc.rows), xr, getNumThreads());
+
+            }
+            else
+                std::cout << "PB";
```

Comment:
what does it mean? Implementation does nothing for non-continuous matrices. It's not acceptable. Please, rewrite the implementation to work properly with all matrices.